lib/bgm.dart

```diff
@@ -48,11 +48,11 @@ class Bgm extends WidgetsBindingObserver {
   ///
   /// It is safe to call this function even when a current BGM track is
   /// playing.
-  Future<void> play(String filename, {double volume}) async {
+  Future<void> play(String filename, {double? volume}) async {
     volume ??= 1;
 
-    if (audioPlayer != null && audioPlayer.state != AudioPlayerState.STOPPED) {
-      audioPlayer.stop();
+    if (audioPlayer != null && audioPlayer!.state != AudioPlayerState.STOPPED) {
```

Comment:
so (for this and 3 places below): I am currently leaning towards assign the field to a local var at the beginning of the method:

```
  final audioPlayer = audioPlayer;
```
that way you get smart cast/type promotion and avoid all the `!`

but I am open to discussing this standard (at least until we get better features on dart to handle this)

Reply:
I think the same about this as I explained on the previous comment, a `!` is less code than declaring a new variable, and since this is used inside a checking, seems just as safe.

Maybe this could be a broader discussion that we could do for our styleguide?

Reply:
Or use `dartlin`'s `let` method :-).

Reply:
yes, I am proposing we discuss this, as maybe there are better options, but having a lot of ! is introducing potential flaws

Reply:
changed to local variable